ci_framework/roles/cifmw_cephadm/tasks/rgw.yml

```diff
@@ -29,6 +29,20 @@
     _hosts: "{{ _hosts|default([]) + [ hostvars[item][ceph_hostname_var] ] }}"
   loop: "{{ groups['edpm'] }}"
 
+- name: Get and save TLS certificate for rgw spec if provided
+  become: true
+  block:
+    - name : Get the certificate content
+      register: slurp_cert
+      ansible.builtin.slurp:
+        src: "{{ cifmw_cephadm_certificate }}"
+
+    - name : Set rgw_frontend_cert for rgw spec
+      set_fact:
+        rgw_frontend_cert: "{{ slurp_cert.get('content', '') | b64decode }}"
+  when: cifmw_cephadm_certificate | length > 0 and
```

Comment:
care to push the `when` after the associated task `name` ? It will be easier to read.